src/header/mod.rs

```diff
@@ -695,6 +707,19 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_eq() {
+        let mut headers1 = Headers::new();
+        let mut headers2 = Headers::new();
+
+        assert_eq!(headers1, headers2);
+
+        headers1.set(ContentLength(11));
+        headers2.set(ContentLength(11));
+
+        assert_eq!(headers1, headers2);
+    }
```

Comment:
Additional tests would be useful:

- `{ Content-Length: 10 }` != `{ Content-Length: 11 }`
- `{ Content-Length: 10 }` != `{ Content-Length: 10, Server: foo }`
- `{ Content-Length: 10 }` != `{ Server: foo }`